pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -869,6 +877,131 @@ func afwGetReleventStatefulsetByKind(cr splcommon.MetaObject, client splcommon.C
+func (podExecClient *IdxcPodExecClient) runPodExecCommand() (string, string, error) {
+	if podExecClient == nil {
+		err := fmt.Errorf("podExecClient is nil")
+		return "", "", err
+	}
+
+	streamOptions := &remotecommand.StreamOptions{
+		Stdin: strings.NewReader(podExecClient.command),
+	}
+
+	return splutil.PodExecCommand(podExecClient.client, podExecClient.targetPodName, podExecClient.cr.GetNamespace(), []string{"/bin/sh"}, streamOptions, false, false)
+}
+
+// isBundlePushComplete checks the status of bundle push
+func (idxcPlayBookContext *IdxcPlayBookContext) isBundlePushComplete() bool {
+	scopedLog := log.WithName("isBundlePushComplete").WithValues("crName", idxcPlayBookContext.cr.GetName(), "namespace", idxcPlayBookContext.cr.GetNamespace())
+
+	stdOut, stdErr, err := idxcPlayBookContext.podExecClient.runPodExecCommand()
+	if err != nil || stdErr != "" {
+		scopedLog.Error(err, "show cluster-bundle-status failed", "stdout", stdOut, "stderr", stdErr)
+		return false
+	}
+
+	if !strings.Contains(stdOut, "cluster_status=None") {
+		scopedLog.Info("Bundle push is still in progress")
+		return false
+	}
+
+	// bundle push is complete
+	scopedLog.Info("Bundle push is complete")
+	return true
+}
+
+// RunPlayBook will implement the following logic(and set the bundle push state accordingly)  -
+// 1. If the bundle push is not in progress, run the logic to push the bundle from CM to indexer peers
+// 2. OR else, if the bundle push is already in progress, check the status of bundle push
+func (idxcPlayBookContext *IdxcPlayBookContext) runPlayBook() error {
+
+	scopedLog := log.WithName("RunPlayBook").WithValues("crName", idxcPlayBookContext.cr.GetName(), "namespace", idxcPlayBookContext.cr.GetNamespace())
+
+	appDeployContext := idxcPlayBookContext.afwPipeline.appDeployContext
+
+	// if the bundle push is already in progress, check the status
+	if appDeployContext.BundlePushStatus.BudlePushStage == enterpriseApi.BundlePushInProgress {
+		podExecClient := idxcPlayBookContext.podExecClient.(*IdxcPodExecClient)
+
+		// change the command string to run the show cluster-bundle-status command for idxc
+		podExecClient.command = fmt.Sprintf(idxcShowClusterBundleStatusStr, podExecClient.adminPwd)
+		if idxcPlayBookContext.isBundlePushComplete() {
+			// set the bundle push status to complete
+			setBundlePushState(afwPipeline, enterpriseApi.BundlePushComplete)
+		} else {
+			scopedLog.Info("bundle push is still in progress, will check back again in next reconcile..")
+		}
+		return nil
+	}
+
+	scopedLog.Info("running command to apply cluster bundle")
+	stdOut, stdErr, err := idxcPlayBookContext.podExecClient.runPodExecCommand()
+	if err != nil || stdErr != "OK\n" {
+		err = fmt.Errorf("error while applying cluster bundle. stdout: %s, stderr: %s, err: %v", stdOut, stdErr, err)
+		scopedLog.Error(err, "failed to apply cluster bundle")
+		return err
+	}
+
+	setBundlePushState(afwPipeline, enterpriseApi.BundlePushInProgress)
+	return nil
```

Comment:
nit: Since you wrote `isBundlePushComplete`  for checking the status, does it make sense to have another function to trigger the bundle push. 
Also, a switch/if-else-if block may serve better, as it is driven by the bundle push status?